src/rowtransforms.jl

```diff
@@ -1,47 +1,50 @@
-struct NormalizeRow{T, S} <: DataAugmentation.Transform
-        normstats::T
-        normcols::S
+struct NormalizeRow{T, S} <: Transform
+    dict::T
+    cols::S
 end
 
-struct Categorify{T, S} <: DataAugmentation.Transform
-        catdict::T
-        categorycols::S
+struct FillMissing{T, S} <: Transform
+    dict::T
+    cols::S
 end
 
-struct FillMissing{T, S} <: DataAugmentation.Transform
-        fmvals::T
-        fmcols::S
-end
-
-function DataAugmentation.apply(tfm::FillMissing, item::TabularItem; randstate=nothing)
-    x = [val for val in item.data]
-    for col in tfm.fmcols
-            idx = findfirst(col .== item.columns)
-            if ismissing(x[idx])
-                x[idx] = tfm.fmvals[col]
-            end
+struct Categorify{T, S}
+    dict::T
+    cols::S
+    function Categorify{T, S}(dict::T, cols::S) where {T, S}
+        for (col, vals) in dict
+            dict[col] = append!([], [missing], collect(skipmissing(Set(vals))))
+        end
+        new{T, S}(dict, cols)
     end
-    x = (; zip(item.columns, x)...)
-    TabularItem(x, item.columns)
 end
 
-function DataAugmentation.apply(tfm::NormalizeRow, item::TabularItem; randstate=nothing)
-    x = [val for val in item.data]
-    for col in tfm.normcols
-            idx = findfirst(col .== item.columns)
-            colmean, colstd = tfm.normstats[col]
-            x[idx] = (x[idx] - colmean)/colstd
-    end
-    x = (; zip(item.columns, x)...)
-    TabularItem(x, item.columns)
+Categorify(dict::T, cols::S) where {T, S} = Categorify{T, S}(dict, cols)
+
+function apply(tfm::NormalizeRow, item; randstate=nothing)
```

Comment:
Is `randstate` an artifact from Python? Or is it part of the DataAugmentation interface? What is its role here?

Reply:
Yeah, I can't see which of these transforms requires an RNG.

Reply:
Yeah, even though `randstate` isn't required for the tabular transformations, I put it there because it was a part of the transformation interface. I think internally for compositions, `apply` is called along with `randstate` args so everything might not work without it.

Reply:
Yeah, it's because of how the dispatch is set up.